react-routing/client/src/App.tsx

```diff
@@ -0,0 +1,27 @@
+import { Routes, Route } from 'react-router-dom';
+import { Header } from './components/Header';
+import { Catalog } from './pages/Catalog';
+import { ProductDetails } from './pages/ProductDetails';
+import { About } from './pages/About';
+import { NotFound } from './pages/NotFound';
+
+export function App() {
+  return (
+    <div>
+      <Routes>
+        <Route path="/" element={<Header />}>
+          <Route index element={<Catalog />} />
+          <Route
+            path="details/:productId"
+            loader={({ params }): string => {
```

Comment:
You don't need this loader prop at all. ProductDetails gets the productId with `useParams`.